packages/auth/src/accessToken/service.ts

```diff
@@ -159,31 +173,49 @@ async function createAccessToken(
 
 async function rotate(
   deps: ServiceDependencies,
-  managementId: string
+  managementId: string,
+  tokenValue: string
 ): Promise<Rotation> {
-  let token = await AccessToken.query(deps.knex).findOne({ managementId })
-  if (token) {
-    await token.$query(deps.knex).delete()
-    token = await AccessToken.query(deps.knex).insertAndFetch({
-      value: crypto.randomBytes(8).toString('hex').toUpperCase(),
-      grantId: token.grantId,
-      expiresIn: token.expiresIn,
-      managementId: v4()
-    })
-    const access = await Access.query(deps.knex).where({
-      grantId: token.grantId
-    })
-    return {
-      success: true,
-      access,
-      value: token.value,
-      managementId: token.managementId,
-      expiresIn: token.expiresIn
+  const trx = await AccessToken.startTransaction()
+  try {
```

Comment:
I prefer :point_down: so we don't have to manage `trx.commit`/`rollback`
```suggestion
  try {
    return await AccessToken.transaction(async trx => {
```